src/main/java/frc/team2412/robot/util/TrapAlign.java

```diff
@@ -30,20 +32,20 @@ public class TrapAlign {
 		// trap that faces source
 		new Pose2d(new Translation2d(12.3, 3.09), Rotation2d.fromDegrees(120)),
 		// trap that faces mid
-		new Pose2d(new Translation2d(10, 4), Rotation2d.fromDegrees(0))
+		// DO THIS ONE FIRST
+		// brute force the X lol
+		new Pose2d(new Translation2d(10.8, 4.10), Rotation2d.fromDegrees(0))
 	};
 
 	private static Command trapAlign(DrivebaseSubsystem drivebaseSubsystem) {
 		Pose2d robotPose = drivebaseSubsystem.getPose();
-		Pose2d trapPose =
-				robotPose.nearest(
-						List.of(
-								(DriverStation.getAlliance().get().equals(Alliance.Blue))
-										? BLUE_TRAP_POSES
-										: RED_TRAP_POSES));
-
+		boolean isBlue;
+		isBlue = DriverStation.getAlliance().get().equals(Alliance.Blue);
```

Comment:
We should add a check for if the optional is empty, otherwise calling `get()` on an empty optional will throw an exception:

```
Error at java.base/java.util.Optional.get(Optional.java:143): Unhandled exception: java.util.NoSuchElementException: No value present
        at java.base/java.util.Optional.get(Optional.java:143)
        at frc.team2412.robot.util.TrapAlign.trapAlign(TrapAlign.java:43)
        at frc.team2412.robot.util.TrapAlign$AlignCommand.initialize(TrapAlign.java:85)
        at edu.wpi.first.wpilibj2.command.CommandScheduler.initCommand(CommandScheduler.java:166)
        at edu.wpi.first.wpilibj2.command.CommandScheduler.schedule(CommandScheduler.java:223)
        at edu.wpi.first.wpilibj2.command.CommandScheduler.schedule(CommandScheduler.java:234)
        at edu.wpi.first.wpilibj2.command.Command.schedule(Command.java:460)
        at edu.wpi.first.wpilibj2.command.button.Trigger$1.run(Trigger.java:69)
        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
        at edu.wpi.first.wpilibj.event.EventLoop.poll(EventLoop.java:37)
        at edu.wpi.first.wpilibj2.command.CommandScheduler.run(CommandScheduler.java:271)
        at frc.team2412.robot.Robot.robotPeriodic(Robot.java:136)
        at edu.wpi.first.wpilibj.IterativeRobotBase.loopFunc(IterativeRobotBase.java:381)
        at edu.wpi.first.wpilibj.TimedRobot.startCompetition(TimedRobot.java:131)
        at edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:365)
        at edu.wpi.first.wpilibj.RobotBase.lambda$startRobot$0(RobotBase.java:433)
        at java.base/java.lang.Thread.run(Thread.java:833)
```

Arguably, this is unlikely to happen in a match, since alliance color is typically set by the time autonomous starts, but probably best to be safe.

Reply:
![image](https://github.com/robototes/Crescendo2024/assets/145405364/e4a1999f-04da-4861-bc46-2dd6ab0bd2cb)
 this good?